File: MacOs/Donezy/ArchiveScreen.swift
import SwiftUI

struct ArchiveScreen: View {
    @ObservedObject var viewModel: HobbyViewModel
    @Environment(\.theme) private var theme

    var body: some View {
        VStack(spacing: 0) {
            ScreenTopBar(title: "Archived trackers") { viewModel.goHome() }
            if viewModel.archivedHobbies.isEmpty {
                Spacer()
                VStack(spacing: 8) {
                    Text("📦").font(.system(size: 48))
                    Text("No archived trackers").font(.system(size: 16, weight: .semibold))
                    Text("Swipe a tracker on the home screen to archive it.")
                        .font(.system(size: 12)).foregroundColor(theme.onSurfaceVariant)
                        .multilineTextAlignment(.center)
                }
                Spacer()
            } else {
                ScrollView {
                    LazyVStack(spacing: 10) {
                        ForEach(viewModel.archivedHobbies) { hobby in
                            row(hobby)
                        }
                    }
                    .padding(16)
                }
            }
        }
    }

    private func row(_ hobby: Hobby) -> some View {
        let cat = categoryFor(hobby.category)
        return HStack(spacing: 10) {
            CategoryAvatar(category: cat, size: 40)
            VStack(alignment: .leading, spacing: 2) {
                Text(hobby.name).font(.system(size: 14, weight: .semibold)).lineLimit(1)
                Text(cat.label).font(.system(size: 11)).foregroundColor(theme.onSurfaceVariant)
            }
            Spacer()
            Button {
                viewModel.restoreHobby(id: hobby.id, name: hobby.name)
            } label: {
                Label("Restore", systemImage: "arrow.uturn.backward").font(.system(size: 14))
            }
            .buttonStyle(.borderedProminent)
        }
        .padding(12)
        .background(theme.surface, in: RoundedRectangle(cornerRadius: 16))
        .overlay(RoundedRectangle(cornerRadius: 16).strokeBorder(theme.onSurfaceVariant.opacity(0.25)))
    }
}

File: Android/app/src/main/java/com/swarnkary/donezy/ArchiveScreen.kt
package com.swarnkary.donezy

import androidx.activity.compose.BackHandler
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.foundation.lazy.items
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.automirrored.filled.ArrowBack
import androidx.compose.material.icons.filled.Restore
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextOverflow
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import kotlinx.coroutines.flow.collectLatest

@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun ArchiveScreen(viewModel: HobbyViewModel) {
    val archived by viewModel.archivedHobbies.collectAsState()
    val snackbarHostState = remember { SnackbarHostState() }

    BackHandler { viewModel.goHome() }

    LaunchedEffect(Unit) {
        viewModel.snackbarEvents.collectLatest { event ->
            val result = snackbarHostState.showSnackbar(
                message = event.message,
                actionLabel = event.action,
                duration = SnackbarDuration.Short
            )
            if (result == SnackbarResult.ActionPerformed) event.onAction?.invoke()
        }
    }

    Scaffold(
        snackbarHost = { SnackbarHost(snackbarHostState) },
        topBar = {
            TopAppBar(
                title = { Text("Archived trackers") },
                navigationIcon = {
                    IconButton(onClick = { viewModel.goHome() }) {
                        Icon(Icons.AutoMirrored.Filled.ArrowBack, contentDescription = "Back")
                    }
                },
                colors = TopAppBarDefaults.topAppBarColors(containerColor = Color.Transparent)
            )
        }
    ) { padding ->
        if (archived.isEmpty()) {
            Box(
                Modifier.fillMaxSize().padding(padding),
                contentAlignment = Alignment.Center
            ) {
                Column(horizontalAlignment = Alignment.CenterHorizontally) {
                    Text("📦", fontSize = 48.sp)
                    Spacer(Modifier.height(8.dp))
                    Text("No archived trackers", style = MaterialTheme.typography.titleMedium,
                        fontWeight = FontWeight.SemiBold)
                    Text("Swipe a tracker on the home screen to archive it.",
                        style = MaterialTheme.typography.bodySmall,
                        color = MaterialTheme.colorScheme.onSurfaceVariant)
                }
            }
            return@Scaffold
        }

        LazyColumn(
            modifier = Modifier.fillMaxSize().padding(padding),
            contentPadding = PaddingValues(16.dp),
            verticalArrangement = Arrangement.spacedBy(10.dp)
        ) {
            items(archived, key = { it.id }) { hobby ->
                val cat = categoryFor(hobby.category)
                OutlinedCard(
                    modifier = Modifier.fillMaxWidth(),
                    shape = MaterialTheme.shapes.large
                ) {
                    Row(
                        Modifier.padding(12.dp),
                        verticalAlignment = Alignment.CenterVertically,
                        horizontalArrangement = Arrangement.spacedBy(10.dp)
                    ) {
                        CategoryAvatar(cat, size = 40)
                        Column(Modifier.weight(1f)) {
                            Text(hobby.name, style = MaterialTheme.typography.titleSmall, fontWeight = FontWeight.SemiBold,
                                maxLines = 1, overflow = TextOverflow.Ellipsis)
                            Text(cat.label, style = MaterialTheme.typography.labelSmall,
                                color = MaterialTheme.colorScheme.onSurfaceVariant)
                        }
                        FilledTonalButton(
                            onClick = { viewModel.restoreHobby(hobby.id, hobby.name) },
                            shape = MaterialTheme.shapes.large
                        ) {
                            Icon(Icons.Default.Restore, null, Modifier.size(16.dp))
                            Spacer(Modifier.width(4.dp))
                            Text("Restore")
                        }
                    }
                }
            }
        }
    }
}
